lib/pending_assigns_reminder.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+class PendingAssignsReminder
+  REMINDED_AT = 'last_reminded_at'
+  REMINDER_THRESHOLD = 2
+
+  def remind(user)
+    newest_topics = assigned_topics(:newest, user)
+    return if newest_topics.size < REMINDER_THRESHOLD
+    oldest_topics = assigned_topics(:oldest, user).where.not(id: newest_topics.map(&:id))
+    assigned_topics_count = assigned_count_for(user)
+
+    title = I18n.t('pending_assigns_reminder.title', pending_assignments: assigned_topics_count)
+
+    PostCreator.create!(
+      Discourse.system_user,
+      title: title,
+      raw: reminder_body(user, assigned_topics_count, newest_topics, oldest_topics),
+      archetype: Archetype.private_message,
+      target_usernames: user.username,
+      validate: false
+    )
+
+    update_last_reminded(user)
+  end
+
+  private
+
+  def assigned_count_for(user)
+    TopicCustomField.where(name: TopicAssigner::ASSIGNED_TO_ID, value: user.id.to_s).count
+  end
+
+  def assigned_topics(type, user)
+    order = type == :newest ? 'DESC' : 'ASC'
+    Topic.joins(:_custom_fields).select(:slug, :id, :fancy_title, 'topic_custom_fields.created_at AS assigned_at')
+      .where('topic_custom_fields.name = ? AND topic_custom_fields.value = ?', TopicAssigner::ASSIGNED_TO_ID, user.id.to_s)
```

Comment:
```suggestion
      .where('topic_custom_fields.name = ? AND topic_custom_fields.value = ?', TopicAssigner::ASSIGNED_TO_ID, user.id)
```

Reply:
If I apply this suggestion, I'll get a type mismatch error since value must be a string.